### PR TITLE
MGMT-3821 Fix errors with ci/prow/assisted-service-aws

### PR DESCRIPTION
**Pull request by lranjbar**
Fix errors with assisted-service-aws after switching to using a custom build root in CI.

### CODE REVIEW

**Comment by lranjbar** [comment]
@YuviGold @djzager After some investigation it turns out that we are not able to test changes in the Dockerfile.assisted-build before merge. I added a couple of things to test for the changes in openshift/release but I'll need this PR merged.

File: setup_prow_ci_env.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/bash
```
Comment (YuviGold) [inline setup_prow_ci_env.sh:1]: This is not a shell script

File: Makefile [hunk 2]
```diff
@@ -293,7 +293,9 @@ deploy-onprem-for-subsystem:
 	export DUMMY_IGNITION="true" && $(MAKE) deploy-onprem
 
 deploy-on-openshift-ci:
+ifeq ("$(wildcard $(shell which kubectl))","")
```
Comment (YuviGold) [inline Makefile:296]: IMO you can remove 
```
ln -s $(shell which oc) $(shell dirname $(shell which oc))/kubectl
```

File: Dockerfile.assisted-service-build [hunk 3]
```diff
@@ -22,3 +22,5 @@ RUN curl -Lo minikube https://storage.googleapis.com/minikube/releases/v1.10.1/m
   && chmod +x minikube && mkdir -p /usr/local/bin/ && install minikube /usr/local/bin/
 RUN curl -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl && chmod +x ./kubectl && mv ./kubectl /usr/local/bin/kubectl
 RUN curl -SL https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.6.0/openshift-client-linux-4.6.0.tar.gz | tar -xz -C /usr/local/bin
+
+RUN chmod 777 -R /go
```
Comment (YuviGold) [inline Dockerfile.assisted-service-build:26]: Maybe using `$GOPATH` to support future directory change

File: Makefile (hunk 2 above)
Comment (lranjbar) [reply Makefile:296]: I'm looking to test with and without the custom build root image so I still need it for that.

File: setup_prow_ci_env.sh (hunk 1 above)
Comment (lranjbar) [reply setup_prow_ci_env.sh:1]: Doh. Need more coffee.

File: Makefile [hunk 4]
```diff
@@ -292,8 +292,13 @@ deploy-onprem:
 deploy-onprem-for-subsystem:
 	export DUMMY_IGNITION="true" && $(MAKE) deploy-onprem
 
+setup-openshift-ci:
+	setup_prow_ci_env.sh
```
Comment (YuviGold) [inline Makefile:296]: There's already `make build-openshift-ci-test-bin` target being called on https://github.com/openshift/release/blob/357ca3f0f913a28534924305b31aa66aefdbd016/ci-operator/config/openshift/assisted-service/openshift-assisted-service-master.yaml#L28-L29
You should use it.

File: setup_prow_ci_env.sh (hunk 1 above)
Comment (djzager) [reply setup_prow_ci_env.sh:1]: My personal preference is for this stuff to go in the Makefile directly. I'll defer to @YuviGold but having to find the make recipe then the shell script being called gives me a headache.

**Comment by lranjbar** [comment]
Combined the last two comments.

**Comment by lranjbar** [comment]
Workaround found with https://github.com/openshift/release/pull/15442 so closing.

**Pull request closed without merging**